modules/systemd/boot.nix

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+
+  # Ghaf configuration flag
+  cfg = config.ghaf.systemd.boot;
+
+  # Import minimal systemd config
+  minimalSystemd = (import ./minimal.nix { inherit pkgs; }).minimalSystemd;
+
+  # Initrd-systemd package configuration 
+  bootPackageConfiguration =  minimalSystemd.override {
+    pname = "boot-systemd";
+    withAcl = true;
+    withCompression = true;
+    withCoredump = true;
+    withCryptsetup = true;
+    withEfi = pkgs.stdenv.hostPlatform.isEfi;
+    withFido2 = true;
+    withKmod = true;
+    withTpm2Tss = true;
+  };
+    
+  # Excluded systemd units for initrd
+  excludedInitrdUnits = [
+      
+    # Targets
+    # "basic.target"
+    # "swap.target"
```

Comment:
We might want `swap`. Minimal host could be allocated minimal RAM mitigated with `swap`. `swap` can be encrypted.
No swap can lead to OoO and poor perf in general.

Reply:
The elements not commented are here the surpressed units, so the ones commented out are included. Also note that this is only for the systemd in initrd (so stage 1 of the boot), not the host systemd configuration or other.